nixos/tests/mattermost/default.nix

```diff
@@ -342,39 +344,54 @@ import ../make-test-python.nix (
         '';
       in
       ''
+        import sys
         import shlex
+        import threading
+        import queue
 
         def wait_mattermost_up(node, site_name="${siteName}"):
+          print(f"wait_mattermost_up({node.name!r}, site_name={site_name!r})", file=sys.stderr)
+          node.wait_for_unit("multi-user.target")
           node.systemctl("start mattermost.service")
           node.wait_for_unit("mattermost.service")
           node.wait_for_open_port(8065)
           node.succeed(f"curl {shlex.quote('${url}')} >/dev/null")
           node.succeed(f"curl {shlex.quote('${url}')}/index.html | grep {shlex.quote(site_name)}")
 
         def restart_mattermost(node, site_name="${siteName}"):
+          print(f"restart_mattermost({node.name!r}, site_name={site_name!r})", file=sys.stderr)
           node.systemctl("restart mattermost.service")
           wait_mattermost_up(node, site_name)
 
         def expect_config(node, mattermost_version, *configs):
+          print(f"expect_config({node.name!r}, {mattermost_version!r}, *{configs!r})", file=sys.stderr)
           for config in configs:
             node.succeed(f"${expectConfig} {shlex.quote(config)} {shlex.quote(mattermost_version)}")
 
         def expect_plugins(node, jq_or_code):
+          print(f"expect_plugins({node.name!r}, {jq_or_code!r})", file=sys.stderr)
           node.succeed(f"${expectPlugins} {shlex.quote(str(jq_or_code))}")
 
         def ensure_post(node, fail_if_not_found=False):
+          print(f"ensure_post({node.name!r}, fail_if_not_found={fail_if_not_found!r})", file=sys.stderr)
           node.succeed(f"${ensurePost} {shlex.quote('${url}')} {1 if fail_if_not_found else 0}")
 
-        def set_config(node, *configs, nixos_version='25.05'):
+        def set_config(node, *configs, nixos_version='${lib.versions.majorMinor lib.version}'):
+          print(f"set_config({node.name!r}, *{configs!r}, nixos_version={nixos_version!r})", file=sys.stderr)
           for config in configs:
             args = [shlex.quote("${setConfig}")]
             args.append(shlex.quote(config))
             if nixos_version:
               args.append(shlex.quote(str(nixos_version)))
             node.succeed(' '.join(args))
 
-        def run_mattermost_tests(mutableToplevel: str, mutable,
-                                 mostlyMutableToplevel: str, mostlyMutable,
+        def switch_to_specialisation(node, toplevel: str, specialisation: str):
+          print(f"switch_to_specialisation({node.name!r}, {toplevel!r}, {specialisation!r})", file=sys.stderr)
+          node.succeed(f"{toplevel}/specialisation/{specialisation}/bin/switch-to-configuration switch || true")
```

Comment:
For a variety of reasons, the config switch can fail for unrelated reasons to Mattermost if we're on unstable, but Mattermost may still restart. We should just wait for it to come up.